### DISCUSSION TITLE
Simultaneous edits on Sharepoint in browser versus local machine

### DISCUSSION THREAD

**Discussion by johnsidoli** [abraunegg/onedrive#2074] [Q&A] [answered]
Starting to work with this a lot for Sharepoint spreadsheets ... Linux Mint w/One Drive and LIbreoffice Calc , but also via browser editing of spreadsheet from sharepoint web.

I've been trying to test and understand what the functionality is... but if I have a spreadsheet open locally in Calc and my colleague has it open in their browser simultaneously... is there anything we need to be careful of? There doesn't seem to be any warning messages, and I can't tell which session takes priority if changes are made?

Last time I checked I couldn't find any guidance on this, and just wondered if anyone had any tips/info/advice that might be pertinent?

**Comment by abraunegg** [comment, marked as answer]
@johnsidoli 
> I've been trying to test and understand what the functionality is... but if I have a spreadsheet open locally in Calc and my colleague has it open in their browser simultaneously... is there anything we need to be careful of?

I would be very careful here about editing the *same* file online and locally at the same time. You could get into a data loss scenario.

The OneDrive API, when dealing with SharePoint specifically, has a number of issues - specifically, that when you upload a modified file, at the end of the commit, the API throws a HTTP 404 error - because SharePoint moved & modified the file. I cant find the specific reference | issue I raised with the API team on this (I am sure I raised this with them .... ), however there has been zero movement on fixing this issue.  If I failed to raise this - then ... poor form on me and that explains why its not been looked at. Additionally I think this is compounded by the SharePoint bug where they add all their own additional metadata to the file post upload - and I think this is where the 404 error that is received is stemming from - which is something (adding the metadata) they have already advised is a *feature* and wont be fixed - thus, the likelihood of this uploading of a modified file to SharePoint not being fixed. 

References:
* SharePoint throws a 404 error on modified file upload: TBA
* SharePoint modifies files post upload: https://github.com/OneDrive/onedrive-api-docs/issues/935 

The way that this client handles this SharePoint and OneDrive API issue, when uploading a locally modified file, the file will be uploaded as a *new* file - and the client does this by *deleting* the online copy that is being replaced before uploading the updated local file - essentially replacing the online copy with a new file & version.

So essentially, if your co-worker is working on that exact same file at the same time you potentially will get into a data loss scenario.

If I were you, I would be doing *whatever* would be possible to ensure that you are not editing files at the same time to avoid any potential scenario that could lead to data loss.

**Comment by johnsidoli** [comment]
Thanks!
